src/fns/constrained_ops.nr

```diff
@@ -294,6 +294,58 @@ pub(crate) fn validate_gt<let N: u32, let MOD_BITS: u32>(lhs: [Field; N], rhs: [
     assert(result_limb == 0);
 }
 
+pub(crate) fn cmp<let N: u32, let MOD_BITS: u32>(lhs: [Field; N], rhs: [Field; N]) -> bool {
+    // so we do... p - x - r = 0 and there might be borrow flags
+    // a - b = r
+    // p + a - b - r = 0
+    let (comp_result, result, carry_flags, borrow_flags) = unsafe { __cmp_remainder(lhs, rhs) };
+    validate_in_range::<_, MOD_BITS>(result);
+
+    let borrow_shift = 0x1000000000000000000000000000000;
+    let carry_shift = 0x1000000000000000000000000000000;
```

Comment:
These are the same value so why not have a single constant? Also this should be named to make it clear how big they are.